### PR TITLE
website/integrations: add Frappe

### PR DESCRIPTION
**Pull request by Infernogeek1**
## Details

Add Frappe OIDC integration 

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

### CODE REVIEW

**Review by dominic-r — CHANGES_REQUESTED** [review]

File: website/integrations/services/frappe/index.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+The following placeholders will be used:
+
+-   `frappe.company` is the FQDN of the Frappe install.
+-   `authentik.company` is the FQDN of the authentik install.
+-   `provider` is the name for the social login provider in Frappe.
+
+## authentik configuration
+
+1. Create a new OAuth2/OpenID Provider under **Applications** > **Providers** using the following settings:
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:27]:
```suggestion
1. Log in to authentik as an admin, and go to the Admin interface,
2. Create a new OAuth2/OpenID Provider under **Applications** -> **Providers** using the following settings:
```

File: website/integrations/services/frappe/index.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    - **Redirect URIs/Origins (RegEx)**:
+        - `https://frappe.company/api/method/frappe.integrations.oauth2_logins.custom/provider`
+    - **Scopes**: `email`, `openid`, `profile`
+    - **Subject mode**: `Based on the Users's username`
+    - **Include claims in id_token**: `True`
+    - Leave everything else as default
+
+    Take note of **Client ID** and **Client Secret** as you will need them later.
+
+2. Create a new Application under **Applications** > **Applications** and assign the created provider.
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:42]:
```suggestion
2. Create a new Application under **Applications** -> **Applications** and assign the provider that you just created.
```

File: website/integrations/services/frappe/index.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        - Access Token URL: `/application/o/token/`
+        - Redirect URL: `/api/method/frappe.integrations.oauth2_logins.custom/provider/`
+        - API Endpoint: `/application/o/userinfo/`
+
+![](./frappe3.png)
+
+        In Client Information:
+        - Auth URL Data: `{"response_type": "code", "scope": "email profile openid"}`
+
+![](./frappe4.png)
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:76]: Is there any final step here? Is there a way to verify that all of the configs were successful?

Seems like an abrupt ending...  ;-)

**Review by tanberry — CHANGES_REQUESTED** [review]
Thanks so much @Infernogeek1 for this contribution! A few nits with phrasing, and a question about how to end the topic, but looks good! Please make any suggested changes you agree with and let's get this merged!

File: website/integrations/services/frappe/index.md (hunk 3 above)
Comment (Infernogeek1) [reply website/integrations/services/frappe/index.md:76]: As for verifying that it worked, I would offer to go to frappe.company and press that **Login with provider**.

**Review by dominic-r — CHANGES_REQUESTED** [review]

File: website/integrations/services/frappe/index.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+The following placeholders will be used:
+
+-   `frappe.company` is the FQDN of the Frappe install.
+-   `authentik.company` is the FQDN of the authentik install.
+-   `provider` is the name for the social login provider in Frappe.
+
+## authentik configuration
+
+1. Log in to authentik as an admin, and go to the Admin interface,
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:27]:
```suggestion
1. Log in to authentik as an admin, and go to the Admin interface.
```

File: website/integrations/services/frappe/index.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    - **Redirect URIs/Origins (RegEx)**:
+        - `https://frappe.company/api/method/frappe.integrations.oauth2_logins.custom/provider`
+    - **Scopes**: `email`, `openid`, `profile`
+    - **Subject mode**: `Based on the Users's username`
+    - **Include claims in id_token**: `True`
+    - Leave everything else as default
+
+    Take note of **Client ID** and **Client Secret** as you will need them later.
+
+3. Create a new Application under **Applications** > **Applications**, pick a name and a slug, and assign the provider that you have just created.
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:43]:
```suggestion
3. Create a new application under **Applications** -> **Applications**, pick a name and a slug, and assign the provider that you have just created.
```

File: website/integrations/services/frappe/index.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+## Frappe configuration
+
+1. From the Frappe main menu, navigate to **Integrations**, then to **Social Login Key**.
+
+Add a new Social login Key using the black button on top right.
+![](./frappe1.png)
+
+2.  Enter the following settings:
+
+        In Client Credentials section:
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:54]:
```suggestion
        In the **Client Credentials** section:
```

File: website/integrations/services/frappe/index.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        - Enable Social Login: Turn the checkmark to the _on_ position.
+        - Client ID: _client-id-from-authentik_
+        - Client Secret: _client-secret-from-authentik_
+
+        In Configuration section:
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:59]:
```suggestion
        In the **Configuration** section:
```

File: website/integrations/services/frappe/index.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        - Sign ups: Allow
+
+![](./frappe2.png)
+
+        In Identity Details section:
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:64]:
```suggestion
        In the **Identity Details** section:
```

File: website/integrations/services/frappe/index.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        - Base URL: `https://authentik.company/`
+        - In Client URLs section:
+        - Authorize URL: `/application/o/authorize/`
+        - Access Token URL: `/application/o/token/`
+        - Redirect URL: `/api/method/frappe.integrations.oauth2_logins.custom/provider/`
+        - API Endpoint: `/application/o/userinfo/`
+
+![](./frappe3.png)
+
+        In Client Information:
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:74]:
```suggestion
        In the **Client Information** section:
```

File: website/integrations/services/frappe/index.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        - Auth URL Data: `{"response_type": "code", "scope": "email profile openid"}`
+
+![](./frappe4.png) 3. Press the black **Save** button on the top right.
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:77]:
```suggestion
    ![](./frappe4.png) 

    3. Click **Save** on the top right.
```

File: website/integrations/services/frappe/index.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+## Verification
+
+1. Go to `https://frappe.company` from Incognito mode.
+2. Press **Login with provider** on the login screen.
```
Comment (tanberry) [inline website/integrations/services/frappe/index.md:82]:
```suggestion
2. Click **Login with provider** on the login screen.
```

**Review by tanberry — CHANGES_REQUESTED** [review]
Thanks so much for this contribution, @Infernogeek1. A couple of nits... you might need to double-check the indentation on some of the images that are within numbered steps; I am not sure everything is aligned there, but the Suggestion box doesn't allow me to put in proper indentations, only spaces.

Let me know if any questions, and as soon as you make these few changes, let's get it merged!

**Comment by Infernogeek1** [comment]
> Thanks so much for this contribution, @Infernogeek1. A couple of nits... you might need to double-check the indentation on some of the images that are within numbered steps; I am not sure everything is aligned there, but the Suggestion box doesn't allow me to put in proper indentations, only spaces.
> 
> Let me know if any questions, and as soon as you make these few changes, let's get it merged!

@tanberry, I tried `make website-watch` and images in numbered steps rendered fine.

**Review by tanberry — APPROVED** [review]
Thanks so much @Infernogeek1 for all the fixes, we'll merge now.

**Comment by Infernogeek1** [comment]
I have dogfooded my own docs and found out I screwed up and included a trailing slash where it should not have been.
It should be
```diff
- Redirect URL: /api/method/frappe.integrations.oauth2_logins.custom/provider/
+ Redirect URL: /api/method/frappe.integrations.oauth2_logins.custom/provider
```
@tanberry

**Comment by tanberry** [comment]
> I have dogfooded my own docs and found out I screwed up and included a trailing slash where it should not have been. It should be
> 
> ```diff
> - Redirect URL: /api/method/frappe.integrations.oauth2_logins.custom/provider/
> + Redirect URL: /api/method/frappe.integrations.oauth2_logins.custom/provider
> ```
> 
> @tanberry

Hi hi, @Infernogeek1 good catch, thanks. Do you want to open a new PR with this fix (since the PR has already been merged), or I can do it, as you wish.

**Pull request merged**